### PR TITLE
Move "Implement selected key stage 3 Teach Computing Curriculum resources" from section 2 to 3

### PR DESCRIPTION
**Pull request by croab**
## Status

* Current Status: WIP
* Review App link(s): ...
* Closes [#2712](https://github.com/NCCE/teachcomputing.org-issues/issues/2712)

## Review progress:

- [x] Browser tested (although could do with further testing)
- [x] Tech review completed

## What's changed?

Moved the affected activity with title "Implement selected key stage 3 Teach Computing Curriculum resources" from its original programme activity grouping (which places it under section 2 in the UI) to its second programme activity grouping (placing it under 3 in the UI). Script then updates the `required_for_completion` attribute on the original grouping from 3 to 2.

## Steps to perform after deploying to production

- Run the method on `ProgrammeActivityGrouping.move_i_belong_section_from_2_to_3` to perform the above changes
- Remove this method in a separate PR

### CODE REVIEW

**Comment by croab** [comment]
@markjs - unsure whether it would be better to run the data changes in the console directly and then just merge the seed changes, or whether the whole thing can just be merged and then the migration method removed after?

File: app/models/programme_activity_grouping.rb [hunk 1]
```diff
@@ -58,6 +58,34 @@ def users_completed(users:)
...
+    raise "Current grouping could not be found" if original_grouping.blank?
+
+    new_grouping = potential_groupings.where.not(id: original_grouping.id).first
+    raise "New grouping could not be determined" if new_grouping.blank?
+
+    programme_activity.update!(programme_activity_grouping_id: new_grouping.id)
+
+    # Update the original grouping to only require 2 completed activities
+    original_grouping.update!(required_for_completion: 2)
+  end
```
Comment (GDTimi) [inline app/models/programme_activity_grouping.rb:87]: Running the method twice flipped it back, but both groups then had required_for_completion set to 2. Might need something to catch this, else can run into an outcome of updating the last group to "Increase girls' engagement by completing at least two activities".

**Review by GDTimi — CHANGES_REQUESTED** [review]
Just left a comment on potentially updating the required_for_completion on the final activity group unintentionally. :)

File: app/models/programme_activity_grouping.rb (hunk 1 above)
Comment (croab) [reply app/models/programme_activity_grouping.rb:87]: Nice, instead of a guard I just ensured that it would run the same every time in the end!

Comment (GDTimi) [reply app/models/programme_activity_grouping.rb:87]: Just re-ran the method, I don't think it has had quite the desired affect, as 'Implement selected key stage 3 Teach Computing Curriculum resources' isn't visible in either group for me afterwards (although the "at least one activity" has remained!).

**Review by GDTimi — CHANGES_REQUESTED** [review]
The 'Implement selected key stage 3 Teach Computing Curriculum resources' activity wasn't visible for me after running the method, unless there is something else I'm doing on my end!

File: app/models/programme_activity_grouping.rb (hunk 1 above)
Comment (croab) [reply app/models/programme_activity_grouping.rb:84]: Should work OK now

Comment (GDTimi) [reply app/models/programme_activity_grouping.rb:84]: Yep, that looks to have addressed it - nice one. 👍

**Review by GDTimi — APPROVED** [review]

**Pull request merged**